src/arrayMethodSort.js

```diff
@@ -4,8 +4,64 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    let sortingFunc;
+
+    // Compare elements as strings if compareFunction is omitted
+    if (compareFunction !== undefined) {
+      sortingFunc = compareFunction;
+    } else {
+      sortingFunc = (a, b) => {
+        if (String(a) === String(b)) {
+          return 0;
+        }
+
+        if (String(a) > String(b)) {
+          return 1;
+        }
+
+        return -1;
+      };
+    }
+
+    function partition(arr, start, end) {
```

Comment:
The partition function is well-implemented, but it's not necessary to declare it inside the sort2 method. Declaring it outside would make the sort2 method cleaner and easier to read.